### ISSUE TITLE
REPL bytes repr is incorrect

### ISSUE THREAD

**Issue by rth** [pyodide/pyodide#1810]
With 0.18 in REPL,

```py
>>> x = b'123'
>>> x
49,50,51  
>>> x.__repr__()
"b'123'"
>>> x.__str__()
"b'123'"
```
### Expected result
```py
>>> x
b'123'
```

I haven't checked the code yet, but I would have expected `__repr__` to be used for displaying REPL output.

**Comment by rth** [comment]
Actually there is the same issue with memoryviews,
```py
>>> array = bytearray('ABC', 'utf-8')
```
### Expected
```py
>>> memoryview(array)
<memory at 0x0000.....>
```
### Obtained in REPL,
```py
>>> memoryview(array)
65,66,67
```

**Comment by rth** [comment]
As far as I can tell the source cause is that we are [calling repr](https://github.com/pyodide/pyodide/blob/5a63152172ce39c94c5bfde66a0b66667b916f18/src/templates/console.html#L92) not on the PyProxy of the byte object but on the JsProxy obtained with `to_js` [here](https://github.com/pyodide/pyodide/blob/5a63152172ce39c94c5bfde66a0b66667b916f18/src/templates/console.html#L40) which indeed has the unexpected repr,
```py
>>> from pyodide import to_js
>>> x_js = to_js(b'123')
>>> type(x_js)
<class 'pyodide.JsProxy'>
>>> str(x_js)
'49,50,51'
```

**Comment by hoodmane** [comment]
The `depth=1` argument should mean that the result is a Javascript Array with one element which should be a `PyProxy` to the result. But this clearly isn't working quite as expected.

**Comment by hoodmane** [comment]
Oh wait this is already resolved in main by #1780. Maybe we should also include that in 0.18.1 .

**Comment by rth** [comment]
Aww, yes, I forgot about that PR. 

> Maybe we should also include that in 0.18.1 .

+1
